exp/internal/controllers/machinepool_controller_test.go

```diff
@@ -302,6 +305,7 @@ func TestReconcileMachinePoolRequest(t *testing.T) {
 	testCases := []struct {
 		machinePool expv1.MachinePool
 		expected    expected
+		withTracker bool
```

Comment:
nit: I'd expect test cases to use tracker by default. So this would be something like `omitTracker` or `forceDeleteNodeFailure` that needs to be explicitly set to true by the test case.

Reply:
well, Instead of init the tracker and then set it to nil better to check if we need it and init it